shared/lib_battery_powerflow.cpp

```diff
@@ -333,22 +355,31 @@ void BatteryPowerFlow::calculateACConnected()
             P_batt_to_grid_ac = P_batt_to_load_ac = 0;
             P_battery_ac = 0;
         }
-        P_pv_to_load_ac = P_pv_ac;
+        if (m_BatteryPower->dischargeOnlyLoadExceedSystem) {
+            P_pv_to_load_ac = P_pv_ac;
 
-        // Excess PV production, no other component meets load
-        if (P_pv_ac >= calc_load_ac)
-        {
-            P_pv_to_load_ac = calc_load_ac;
-            P_fuelcell_to_load_ac = 0;
-            P_batt_to_load_ac = 0;
+            // Excess PV production, no other component meets load
+            if (P_pv_ac >= calc_load_ac)
+            {
+                P_pv_to_load_ac = calc_load_ac;
+                P_fuelcell_to_load_ac = 0;
+                P_batt_to_load_ac = 0;
 
-            // discharging to grid
-            P_pv_to_grid_ac = P_pv_ac - P_pv_to_load_ac;
-            P_fuelcell_to_grid_ac = P_fuelcell_ac;   
+                // discharging to grid
+                P_pv_to_grid_ac = P_pv_ac - P_pv_to_load_ac;
+                P_fuelcell_to_grid_ac = P_fuelcell_ac;
+            }
+            else {
+                P_fuelcell_to_load_ac = std::fmin(P_fuelcell_ac, calc_load_ac - P_pv_to_load_ac);
+                P_batt_to_load_ac = std::fmin(P_battery_ac - P_system_loss_ac, calc_load_ac - P_pv_to_load_ac - P_fuelcell_to_load_ac);
+            }
         }
         else {
-            P_fuelcell_to_load_ac = std::fmin(P_fuelcell_ac, calc_load_ac - P_pv_to_load_ac);
-            P_batt_to_load_ac = std::fmin(P_battery_ac - P_system_loss_ac, calc_load_ac - P_pv_to_load_ac - P_fuelcell_to_load_ac);
+            P_batt_to_load_ac = P_battery_ac;
```

Comment:
I was trying to avoid doing batt to grid here, as that increases the scope to the dispatch code. The fmin addition is a good idea, I'll get that in the next push.